src/dss/remove_notebook.py

```diff
@@ -0,0 +1,32 @@
+from lightkube import Client
+from lightkube.core.exceptions import ApiError
+from lightkube.resources.apps_v1 import Deployment
+from lightkube.resources.core_v1 import Service
+
+from dss.config import DSS_NAMESPACE
+from dss.logger import setup_logger
+
+# Set up logger
+logger = setup_logger("logs/dss.log")
+
+
+def remove_notebook(name: str, lightkube_client: Client) -> None:
+    """
+    Removes a Notebook server from the Kubernetes cluster.
+
+    Args:
+        name (str): The name of the notebook server.
+        lightkube_client (Client): The Kubernetes client.
+    """
+
+    try:
+        lightkube_client.delete(res=Deployment, name=name, namespace=DSS_NAMESPACE)
+        lightkube_client.delete(res=Service, name=name, namespace=DSS_NAMESPACE)
```

Comment:
Just a drive-by comment on this that came up in #43 (specifically [here](https://github.com/canonical/data-science-stack/pull/43/files/627aff24391098bdb7a0506f3b11f173735eae7c#r1523105906))...  I realized create-notebook was creating its own deletion function when things timed out, which is not good.  

I wonder if we should use an `ownerReference` on the `Service` so that it will automatically get cleaned up if the notebook `Deployment` is deleted

Reply:
@kimwnasptd probably has a better sense of what is appropriate.  But I wouldn't block this PR on the choice.  If this PR gets to the point where everything else is reviewed and ready to go, lets merge with whatever you choose and we can always change it after.

Reply:
I believe both the labels and the `ownerReference` will be helpful here. It looks like in this PR you could consider deleting the resources with the `dss.configDSS_CLI_MANAGER_LABELS` label. Perhaps in a separate PR you could also add an `ownerReference` to the `Service` so it only gets deleted IFF the `Deployment` gets deleted, as @ca-scribner suggests.

Reply:
Thanks for the ping! `ownerReferences` is something I've seen in operator pattern, in which an object owns other objects (CR owns underlying objects).

I would suggest to not go with this road for now, as this is more of an indicator we need a K8s controller. And in this case it means the CLI is this controller.

Reply:
I'm okay with what @kimwnasptd mentions, if we consider it for the future, sounds like it can be added in a separate PR. I wonder if you are still considering the labels, though @NohaIhab.

Reply:
current way lgtm

Reply:
@DnPlas what's your suggestion on using labels? I think what you're suggesting would apply to #75 to remove all the resources managed by DSS, but I'm not sure how it applies here. Can you elaborate?

Reply:
I was asking because you mentioned it in an earlier comment, but if that doesn't apply here, there is nothing else to discuss (: